src/frontend/js/widgets/Dashboard/components/DashboardItem/Enrollment/ProductCertificateFooter/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { FormattedMessage } from 'react-intl';
+import PurchaseButton from 'components/PurchaseButton';
+import { Icon, IconTypeEnum } from 'components/Icon';
+import { CourseLight, Order, Product, ProductType } from 'types/Joanie';
+import { useOrders } from 'hooks/useOrders';
+import { CourseProductProvider } from 'contexts/CourseProductContext';
+import DownloadCertificateButton from 'components/DownloadCertificateButton';
+import { useCertificate } from 'hooks/useCertificates';
+import CertificateStatus from '../../CertificateStatus';
+
+const messages = {
+  buyProductCertificateLabel: {
+    id: 'components.ProductCertificateRow.buyProductCertificateLabel',
+    description: 'Label on the enrollement row that propose to buy a product of type certificate',
+    defaultMessage: 'A exam that provide a certificate can be purchase for this course.',
+  },
+  downloadProductCertificateLabel: {
+    id: 'components.ProductCertificateRow.downloadProductCertificateLabel',
+    description: 'Label on the enrollement row that propose to download a certificate',
+    defaultMessage: 'A certificate is available for download.',
+  },
+  pendingProductCertificateLabel: {
+    id: 'components.ProductCertificateRow.pendingProductCertificateLabel',
+    description: 'Label on the enrollement when a product of type certificate have been bought',
+    defaultMessage: 'Finish this course to obtain your certificate.',
+  },
+};
+
+interface ProductCertificateRowProps {
+  course: CourseLight;
+  product: Product;
+}
+
+const ProductCertificateFooter = ({ course, product }: ProductCertificateRowProps) => {
+  if (product.type !== ProductType.CERTIFICATE) {
+    return null;
+  }
+  const { items: orders } = useOrders({ product: product.id, course: course.code });
+  // FIXME(rlecellier): what to do if we have multiple orders ?
+  const order: Order | null = orders && orders.length ? orders[0] : null;
+  const { item: certificate } = useCertificate(order?.certificate);
+
+  return (
+    <CourseProductProvider courseCode={course.code} productId={product.id}>
+      <div className="dashboard-item__course-enrolling__infos">
+        <div className="dashboard-item__block__status">
+          <Icon name={IconTypeEnum.CERTIFICATE} />
+          {order ? (
+            <>
+              {product.certificate_definition.title + '. '}
+              <CertificateStatus certificate={certificate} productType={product.type} />
+            </>
+          ) : (
+            <FormattedMessage {...messages.buyProductCertificateLabel} />
+          )}
+        </div>
+        {order ? (
+          order.certificate && (
+            <DownloadCertificateButton
+              className="dashboard-item__button"
+              certificateId={order.certificate}
+            />
+          )
+        ) : (
+          <PurchaseButton className="dashboard-item__button" product={product} />
```

Comment:
To display the `PurchaseButton`, the related course run state must be future or ongoing. We do not want to sell certificate for archived course run.

Reply:
It's more than just the button, we should not display the `ProductCertificateFooter` for enrollement of a old course run with product but no order.
I'll return null on this case.

Reply:
Nope because if the user has purchased the certificate product, we want to display this component to allow user to download its certificate

Reply:
Should i use `courseRun.start` or a `state`  ?
```js
export enum Priority {
  ONGOING_OPEN,
  FUTURE_OPEN,
  ARCHIVED_OPEN,
  FUTURE_NOT_YET_OPEN,
  FUTURE_CLOSED,
  ONGOING_CLOSED,
  ARCHIVED_CLOSED,
  TO_BE_SCHEDULED,
}
```

Reply:
state.priority is the right property to use I think. It should be less than ˋARCHIVED_CLOSED` and should also exclude ˋACHIVED_OPEN`.